### DISCUSSION TITLE
[Question]: aspect ratio changes

### DISCUSSION THREAD

**Discussion by ElectronicBattle** [blakeblackshear/frigate#12968] [Ask A Question] [answered]
### What is your question:

hello All

Having read the documentation for 0.14, I thought I knew why my camera's displayed aspect ratio in 0.14 changes.  I thought it was because the camera stream is switched from low to high res as the UI detects motion and presents a higher quality. Here is an example for my cameras:

![16x9](https://github.com/user-attachments/assets/9a01bfef-4a36-4e79-b770-94e263791aa2)
![4x3](https://github.com/user-attachments/assets/7e8d0ef7-72c6-47c4-9750-70a498182642)

However, I have two identical aspect ratios being presented to Frigate:
HD = 1920x1080 = 16:9 = 1.778
SD = 640x360 =  16:9 = 1.778

(ffprobe confirms these values).

My config asks for the SD and HD streams and it specifies the detection size as being that of the SD stream. I expect I have misunderstood yet again. Please can you explain it to me?

My [config](https://pastebin.com/bY2iHiZr)  if needed/interested: this is 0.14.

**Comment by hawkeye217** [comment]
It looks like there's still a mismatch. The high res stream doesn't look to be 16:9. What does ffprobe from inside the Frigate container of the go2rtc stream show?

`ffprobe rtsp://127.0.0.1:8554/upper_patio`

**Reply by ElectronicBattle** [reply]
Ahh, interesting - thanks !

During tomorrow, I'll try to find out (unlike my 0.13 instance, i'm running 0.14 via Docker compose and I have only just learned what to type to get the container up and detached, and then down  or recreated. I need to find out how to get "inside" the mounted container and, as you know,  I'm a slow learner!).

**Comment by hawkeye217** [comment]
You can get a shell inside the container by running:

`docker exec -it <container_name> /bin/bash`

One thing you could try in the meantime is specifying explicit dimensions for your stream in your go2rtc line:

```
- ffmpeg:rtsp://192.168.1.123:554/media=0/channel=1&level=0/track0#video=h264#width=1920#height=1080#audio=copy#hardware
```

**Reply by ElectronicBattle** [reply]
That's surprisingly simple to do in docker (compose)!

I get told 

`[rtsp @ 0x55bca2f45140] method SETUP failed: 461 Unsupported transport`

and I'll experiment with that modified string as well, reporting back here when I get the answer. Great - thanks again!

**Reply by hawkeye217** [reply]
There's no other information that gets printed other than the 461 error?

**Reply by ElectronicBattle** [reply]
It's the only one which I think is salient, but here's a quick screenshot (to save space):

![image](https://github.com/user-attachments/assets/ec99be36-4aec-4187-8f73-f984da6f05c7)

**Reply by hawkeye217** [reply]
It looks like it's timing out. What happens if you let it run for a bit? It should eventually produce something like this, if the stream is up and running:

```
Input #0, rtsp, from 'rtsp://127.0.0.1:8554/camera':
  Metadata:
    title           : go2rtc/1.9.2
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Video: h264 (Main), yuvj420p(pc, bt470bg/bt470bg/bt709, progressive), 2688x1520, 15 fps, 15 tbr, 90k tbn
  Stream #0:1: Audio: aac (LC), 16000 Hz, mono, fltp
```

**Comment by ElectronicBattle** [comment]
Ahh yes, leaving it for ten seconds did the trick:

```
Input #0, rtsp, from 'rtsp://127.0.0.1:8554/upper_patio':
  Metadata:
    title           : go2rtc/1.9.2
  Duration: N/A, start: 0.051733, bitrate: N/A
  Stream #0:0: Video: h264 (High), yuv420p(progressive), 1920x1080, 19.33 fps, 19.33 tbr, 90k tbn
```

I have also "hard set" the aspect ratio as per your line 

`- ffmpeg:rtsp://192.168.1.123:554/media=0/channel=1&level=0/track0#video=h264#width=1920#height=1080#audio=copy#hardware`
`- ffmpeg:rtsp://192.168.1.123:554/media=0/channel=7&level=0/track0#video=h264#width=1920#height=1080#audio=copy#hardware`

for both of my cameras but I still see the aspect ratio switching happening.

**Reply by hawkeye217** [reply]
Thanks. What does ffprobe show for this?

`ffprobe rtsp://192.168.1.123:554/media=0/channel=1&level=0/track0`

and this

`ffprobe rtsp://192.168.1.123:554/media=0/channel=7&level=0/track0`

**Comment by ElectronicBattle** [comment]
Hello - sorry for the delay. Inside the docker container I had to escape the strings in single quotes (otherwise I get 404):
`ffprobe 'rtsp://192.168.1.123:554/media=0/channel=1&level=0/track0'` 
and also 
`ffprobe 'rtsp://192.168.1.123:554/media=0/channel=7&level=0/track0'` 
both give me
```
title           : Jabsco NVR Stream
Duration: N/A, start: 0.000000, bitrate: N/A
Stream #0:0: Video: hevc (Main), yuvj420p(pc), 1920x1080, 16 fps, 19.42 tbr, 90k tbn
Stream #0:1: Audio: pcm_mulaw, 8000 Hz, 1 channels, s16, 64 kb/s

```
(double-checked OK - I _was_ inside the docker container)

**Reply by hawkeye217** [reply]
I'm not sure what's causing the aspect mismatch, then.

If you go to the full Live view for an individual cam, what aspect is used?

**Reply by ElectronicBattle** [reply]
Let me check (I'm doing some setting up of docker with dockge inside TrueNAS in a way which I think is a bit more "bare metal") and I'll be right back!

**Reply by ElectronicBattle** [reply]
Live view:
![image](https://github.com/user-attachments/assets/ff64fcc8-f1af-4a16-b80a-09b6f5ba6663)
both camera displays keep flicking between the two aspects shown originally.

If I change to "all cameras"
![image](https://github.com/user-attachments/assets/983e9fa6-0734-4da6-857a-46b403830e9b)
again, both camera displays keep flicking between the two aspects shown originally.

If I set up a camera group, they also flick between states. 

Should I try to collect some statistical data to  find out e.g. the exact circumstances which make them change, how often they do so, what's the corelation between event and change?

**Reply by hawkeye217** [reply]
Click on an individual camera from the dashboard. What does the single camera Live view look like? 16:9 or 4:3?

**Reply by ElectronicBattle** [reply]
This is what an individual camera looks like when i click on it in Live view:
![image](https://github.com/user-attachments/assets/c9fab23d-93ff-4beb-bf11-83ddb1c7fadd)

In other words, it's a 4:3 presentation in a 4:3 "aperture" (whereas my original pair of screenshots were a 16:9 in a 16:9 , and a 4:3 inside a 16:9 with the necessary black bars).

**Reply by hawkeye217** [reply]
So the browser seems to think the native resolution of the camera is 4:3, then.

**Reply by hawkeye217** [reply, marked as answer]
I think this is likely something that has to do with your particular setup and how your NVR is feeding camera streams to Frigate.

With the way you have it set up now using go2rtc, the browser itself is what's determining the video's width and height, based on the video metadata. The single camera live view shows this is in 4:3, not 16:9.

If you want to avoid aspect ratio flickering, I'd suggest setting your `detect` resolution to 640x480 (not 640x360).

**Reply by ElectronicBattle** [reply]
OK - I'll do some experiments. I note the detect resolution which I shall also play with; I now have a sense of direction for further tests.  It hadn't occurred to me that the browser might be causing some of the formatting.

Excellent work and great help again - thanks!
